src/ngraph/runtime/cpu/op/convert_layout.cpp

```diff
@@ -33,7 +36,14 @@ shared_ptr<Node>
     {
         throw ngraph_error("Incorrect number of new arguments");
     }
-    return make_shared<ConvertLayout>(new_args.at(0), output_layout);
+
+    auto copy = make_shared<ConvertLayout>(new_args.at(0), output_layout);
+
+    if (m_op_annotations)
+      // Copy annotations
+      copy->set_op_annotations(m_op_annotations->clone());
```

Comment:
I don't like the idea of copying the op annotations here. The purpose of `copy_with_new_args` is to produce a new node that is semantically equivalent to the old node, up to reassigning of the inputs. If there is some semantically relevant information being stored in the op annotations, it should be moved out of the annotations.

Alternatively, if we want to change the contract so that it's part of `copy_with_new_args`'s job to preserve annotations, we should refactor more broadly so that all nodes behave this way. Inconsistent treatment of the annotations is risky, IMHO.

Reply:
Thanks Adam. I don't have a strong opinion here but I think things can get buggy if we only do a selective copy of the node information. Propagating annotations outside of this call might also be error prone. I would lean towards your second suggestion but I'll wait for more feedback.